### ISSUE TITLE
Managed snapping (ListView, ScrollViewer) uses hard-coded delay out of sync with inertia

### ISSUE THREAD

**Issue by takla21** [unoplatform/uno#10891]
### Current behavior

While using the uno implementation for item snapping, it takes time before snap. 

```
Uno.UI.FeatureConfiguration.NativeListViewBase.UseNativeSnapHelper = false;
```

![android_slow](https://user-images.githubusercontent.com/17461593/210636544-c177c7ce-72f1-4527-a009-934241fbffaa.gif)

### Expected behavior

It should be a bit faster.

![ios_smooth](https://user-images.githubusercontent.com/17461593/210637140-b85c66ca-46a4-40b4-b45e-41e9a49cda5c.gif)

### How to reproduce it (as minimally and precisely as possible)

- Download this sample [SnappingListView_slow.zip](https://github.com/unoplatform/uno/files/10347025/SnappingListView_slow.zip)
- Deploy on android
- Swipe through items

### Workaround

N/A

### Works on UWP/WinUI

N/A

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

4.6.39

### Affected platforms

Android

### IDE

Visual Studio 2022

### IDE version

17.4.3

**Comment by Xiaoy312** [comment]
yeah, so in order to get rid of the unoplatform/nventive-private#421 infinite oscillating bug
i added the opt-in option to disable the native snapping (`FeatureConfiguration.NativeListViewBase.UseNativeSnapHelper`), which were fighting with out implementation
by doing so, it had also turned off the scrolling velocity/momentum that comes with it

**Comment by jeromelaban** [comment]
@takla21 try this out and let us know, we'll otherwise close the issue. Thanks!

**Comment by Xiaoy312** [comment]
@jeromelaban they already this applied this UseNativeSnapHelper=false to fix another bug
however, by doing so, it also disabled scrolling momentum which they desired to keep (hence this issue)

**Comment by takla21** [comment]
@Xiaoy312 @jeromelaban So to summarize what was discussed here, if the uno snap implementaion, is there a way we can make it smoother or it's something we lost by disabling scrolling momentum?

**Comment by jeromelaban** [comment]
@dr1rrb can help

**Comment by Xiaoy312** [comment]
#11962 addresses the some irks with the LinearSnapHelper that I've noted in my last comment
which is why we should reevaluate once that is merged

note: we would need to update the code, replacing SnapPointsSnapHelper with LinearSnapHelper in 
https://github.com/unoplatform/uno/blob/4.8.15/src/Uno.UI/UI/Xaml/Controls/ListViewBase/NativeListViewBase.SnapPoints.Android.cs#L20

**Comment by dr1rrb** [comment]
With #11962 and with `Uno.UI.FeatureConfiguration.NativeListViewBase.UseNativeSnapHelper = true;` (i.e. the default value) in the startup of the app, the sample is working pretty well (even if I increase the number of items to force the LV to virtualize items). 

So I think this is no longer "urgent" and as it will require a significant amount of work, I think we can postpone this. Opinion @jeromelaban @takla21 @Xiaoy312  ?

TODO: We need to remove the hard-coded delay and configure the inertia end on a snappoint - currently the idea was to wait the given delay before requesting the snapping in order to let the inertia slow down a bit before aborting it and starting the "scroll to" animation at a fairly equivalent speed)

**Comment by dr1rrb** [comment]
@takla21 did you tried with some inertia? Indeed with delay set to 0 it works well if you release the move pointer in the center of the screen, wait a few and then release. However if you "swipe" the screen you might have some weird animations, so I would recommend you, once https://github.com/unoplatform/uno/pull/11962 has been merged, to use the native snapping (i.e. set `Uno.UI.FeatureConfiguration.NativeListViewBase.UseNativeSnapHelper = true;`).

**Comment by takla21** [comment]
@dr1rrb Here's a screenrecording from the sam

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/17461593/236871780-634bdee9-530b-4e7b-9ee4-863ca63ec6f8.gif)

So it seems that it will wait until it's done moving before doing the snap thing. Which makes it a bit robotic in the end, but yeah we're still going with it for now. I'll revisit using the native one once it's ready

Here's the previous sample updated to latest uno version (just in case)
[SnappingListView_new.zip](https://github.com/unoplatform/uno/files/11422703/SnappingListView_new.zip)

**Comment by MartinZikmund** [comment]
@dr1rrb is this also applicable to Skia?

**Comment by MartinZikmund** [comment]
# Repro generator

This issue was classified as not reproducible with a code sample or test.

**Reason:** This is an Android-specific scrolling/snapping timing issue with managed snap points. It involves physical scroll inertia behavior that requires device interaction and cannot be meaningfully tested in a Skia Desktop runtime test.
